File: Backend/Services/Service.System/Models/FunctionModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace Service.System.Models
{
    public class FunctionModel
    {
        [MaxLength(20)]
        public string Code { get; set; }

        [MaxLength(50)]
        public string Name { get; set; }

        [MaxLength(255)]
        public string Url { get; set; }

        [MaxLength(255)]
        public string Icon { get; set; }

        [MaxLength(20)]
        public string ParentCode { get; set; }

        [Required]
        public int Precedence { get; set; }

        [MaxLength(20)]
        public string ModuleCode { get; set; }

        public List<FunctionCommandModel> Commands { get; set; } = new List<FunctionCommandModel>();
    }
}

File: FontEnd/erp/src/app/core/models/function.model.ts
import { FunctionCommandViewModel } from '../models/function-command.model';

export interface FunctionViewModel {
  code: string;
  name: string;
  url: string;
  icon: string;
  parentCode: string;
  precedence: number;
  moduleCode: string;
  commands: FunctionCommandViewModel[];
}
